### PR TITLE
chat: Handle code block input/display

### PR DESCRIPTION
**Pull request by patosullivan**
For #1283 

We need a way to differentiate inline code formatting vs code block formatting in the toolbar, @urcades . For now I used a right caret to indicate inline.

Video demo:

https://user-images.githubusercontent.com/1221094/204890576-d2c25220-e3ce-47bd-8eb7-0210f9e7170f.mov

### CODE REVIEW

**Comment by urcades** [comment]
Hi @patosullivan — just updated our icon set with an additional item for "Code Block":

https://www.figma.com/file/YLjErBShApNlWpE44nqnbC/Indigo-FW2022?node-id=89%3A1848&t=wdtai91b9AJSQa7a-1

Let me know if this works when dropped in, can tweak as necessary!

**Review by urcades — DISMISSED** [review]
Apart from integrating a new/more distinct icon for "code block" input, I'd recommend we additionally increase the code block's internal margin/padding between the code text and the boundaries of the gray background it's contained within:

Screenshot 2022-11-30 at 3 59 27 PM

In the sketch above, the indigo-spec padding is set to 16p horizontal margin and 8p vertical margin in order to visually "center" the text within the container, ignoring the right-side margin as I assume we're allowing for horizontal overflow (no wrapping) of the text. We can use whichever tailwind stand-ins for the measurements indicated here.

**Comment by patosullivan** [comment]
This actually builds fine, but vitest is not ignoring the ts error so the tests fail. Looking into it.

**Comment by urcades** [comment]
I would not mind at all — here's a link to the 24px version:

https://www.figma.com/file/YLjErBShApNlWpE44nqnbC/Indigo-FW2022?node-id=89%3A1849&t=JE8dAqZkeZBqKrzi-1

**Comment by arthyn** [comment]
still seem to be missing bottom padding:
Screen Shot 2022-12-01 at 1 04 24 PM

**Comment by patosullivan** [comment]
> still seem to be missing bottom padding: Screen Shot 2022-12-01 at 1 04 24 PM

can't repro this, are you sure you pulled the latest?

**Comment by arthyn** [comment]
@patosullivan yep this is on your branch latest

File: ui/src/chat/ChatContent/ChatContent.tsx [hunk 1]
```diff
@@ -122,6 +124,14 @@ export function InlineContent({ story }: InlineContentProps) {
     );
   }
 
+  if (isBlockCode(story)) {
+    return (
+      <pre className="bg-gray-50 px-4">
```
Comment (arthyn) [inline ui/src/chat/ChatContent/ChatContent.tsx:129]: Seems like this should be `rounded bg-gray-50 py-2 px-4` and no need to do the checking in ChatContent

Comment (patosullivan) [reply ui/src/chat/ChatContent/ChatContent.tsx:129]: It *should* work like that, but there's a bug in tiptap that creates multiple lines for code blocks:
https://github.com/ueberdosis/tiptap/issues/1585

**Comment by patosullivan** [comment]
fyi @urcades it looks like this: 
image

**Review by arthyn — APPROVED** [review]
lgtm!

**Comment by urcades** [comment]
@patosullivan I was playing around with this a bit and I think it might help to re-order the options in this formatting bar such that all "inline" functions are grouped, and "block"-creating functions are grouped:

Screenshot 2022-12-05 at 3 49 50 PM

I definitely felt a little bit of visual imbalance here, and it helps somewhat to bundle the larger/more square-looking functions with one another. 

In the example above, I think it also helps to add a little bit of space between each group of icons — this lessens the amount of shape/size contrast we're seeing in the screenshot you took.

**Pull request merged**